apis/addonconfigs/cni/v1alpha1/antreaconfig_types.go

```diff
@@ -258,11 +260,62 @@ type AntreaFeatureGates struct {
 
 // AntreaConfigStatus defines the observed state of AntreaConfig
 type AntreaConfigStatus struct {
+	// Message to indicate failure reason
+	// +kubebuilder:validation:Optional
+	Message string `json:"message,omitempty"`
 	// Reference to the data value secret created by controller
 	// +kubebuilder:validation:Optional
 	SecretRef string `json:"secretRef,omitempty"`
 }
 
+type AntreaNsx struct {
+	// enable indicates whether nsxt adapter shall be enabled in the cluster
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	Enable bool `json:"enable,omitempty"`
```

Comment:
We need to implement either:

- A mutating webhook for AntreaConfig to block updating AntreaNsx.Enable and BootstrapFrom.
or
- Register, deregister dynamically (create/delete NSXSA, ProviderSA) dynamically according to AntreaNsx.Enable and BootstrapFrom changes.

Reply:
implemented 2, is item 1 also needed?

Reply:
@edwardbadboy curious about the necessity of the webhook. If 2 is enough, I think we can close this comment